### PR TITLE
Adds support for combined + ASi support

### PR DESCRIPTION
**Pull request by objectivecosta**
Adds support for combined + ASi support

### CODE REVIEW

File: ios.toolchain.cmake [hunk 1]
```diff
@@ -334,6 +335,30 @@ elseif(PLATFORM_INT STREQUAL "OS64COMBINED")
   else()
     message(FATAL_ERROR "Please make sure that you are running CMake 3.14+ to make the OS64COMBINED setting work")
   endif()
+elseif(PLATFORM_INT STREQUAL "OS64COMBINED_ASI")
+  set(SDK_NAME iphoneos)
+  if(MODERN_CMAKE)
+    if(NOT ARCHS)
+      if (XCODE_VERSION_INT VERSION_GREATER 10.0)
```
Comment (kambala-decapitator) [inline ios.toolchain.cmake:342]:
```suggestion
      if (XCODE_VERSION_INT VERSION_GREATER_EQUAL 12.0)
```

ability to build for arm64 simulator appeared only in Xcode 12

**Comment by hjmallon** [comment]
I don't think `lipo` can join arm64-device and arm64-simulator binaries, does this work?

**Comment by kambala-decapitator** [comment]
> I don't think `lipo` can join arm64-device and arm64-simulator binaries, does this work?

I thought the intention of the PR is to produce fat simulator builds, but two arm64 slices definitely can't be combined with `lipo`

**Comment by hjmallon** [comment]
```cmake
set(CMAKE_XCODE_ATTRIBUTE_ARCHS[sdk=iphoneos*] "arm64")
set(CMAKE_XCODE_ATTRIBUTE_ARCHS[sdk=iphonesimulator*] "x86_64 arm64")
set(CMAKE_XCODE_ATTRIBUTE_VALID_ARCHS[sdk=iphoneos*] "arm64")
set(CMAKE_XCODE_ATTRIBUTE_VALID_ARCHS[sdk=iphonesimulator*] "x86_64 arm64")
```

It looks to me like it is trying to use `CMAKE_IOS_INSTALL_COMBINED` to make a `arm64-device/arm64-sim/x86_64-sim` combined binary. Also the name `OS64COMBINED_ASI` would match that.

I don't think this can work, unless CMake upstream integrated some fairly complete version of xcframework support (i.e. super fat structures with separate simulator and device binaries inside).

Fat arm64/x86_64 simulator builds is possible though (i.e. `SIMULATORUNIVERSAL2` or `SIMULATORARM64X8664` or something).

**Comment by objectivecosta** [comment]
Sorry for the delay... @hjmallon is correct, I wanted to create a fat version with everything possible. I think it is doable, don't know if .xcframework is needed.

**Comment by leetal** [comment]
Closed in favour of #178

**Pull request closed without merging**